### PR TITLE
Android Config for GitHub Actions

### PR DESCRIPTION
**Pull request by noisypigeon**
### Description

This PR implements the Android E2E tests check from CircleCI to GitHub Actions. Later PR will implement caching and other optimizations.

### Other changes

- This is replacing the CircleCI config. As long as it does the same thing and passes it doesn't need to be tested further.
- Setup retries on detox's launchApp and reloadReactNative
- Updated Detox and added [ts-retry-promise](https://www.npmjs.com/package/ts-retry-promise) as a dev dependency to enable the retries on launchApp and reloadReactNative.

### Tested

This is replacing the CircleCI config. As long as it does the same thing and passes it doesn't need to be tested further.

### How others should test

No testing needed, just need to verify it passes.

### Backwards compatibility

GitHub Actions will be replacing CircleCI.

### CODE REVIEW

**Comment by MuckT** [comment]
@noisypigeon - In running these tests I am only seeing one Junit report in Github Actions otherwise it looks like the android and iOS tests are passing.

**Comment by noisypigeon** [comment]
YAY! Looks great. I need to provision the other two machines (will do so on Monday) and then this is ready to be merged.

**Comment by noisypigeon** [comment]
Machines one and two are provisioned. Provisioning the third today.

File: packages/mobile/e2e/init.js [hunk 1]
```diff
@@ -1,10 +1,15 @@
 const detox = require('detox')
 import { setDemoMode } from './src/utils/utils'
+import { launchApp } from './src/utils/retries'
 
 beforeAll(async () => {
+  // This is a hack to ensure android app state stays correct. Causes the app to reinstall for every spec
```
Comment (etuleu) [inline packages/mobile/e2e/init.js:6]: interesting! I wonder if this was an actual bug with the app state when the app gets restarted or just a detox issue?

**Review by etuleu — APPROVED** [review]
awesome!

File: packages/mobile/e2e/init.js (hunk 1 above)
Comment (MuckT) [reply packages/mobile/e2e/init.js:6]: I thought it was just a detox issue, but perhaps it is a bug. I will investigate this locally.

Comment (MuckT) [reply packages/mobile/e2e/init.js:6]: Locally the tests are passing, so I believe it to just be a CI issue specific to Android.

**Pull request merged**